### PR TITLE
CUMULUS-2439: Add CMR search client config to CreateReconciliationReport lambda

### PR DESCRIPTION
**Pull request by jennyhliu**
**Summary:** Summary of changes

Addresses [CUMULUS-2439: create-reconciliation-report not retrieving all expected collections from CMR](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2439)

## Changes

* Added CMR search client setting to the CreateReconciliationReport lambda function.
* Added `cmr_search_client_config` tfvars to the archive and cumulus terraform modules.
* Updated CreateReconciliationReport lambda to search CMR collections with CMRSearchConceptQueue.

## Test:
* Configure cmr_search_client_config in terraform.tfvars, e.g.
cmr_search_client_config = {
  create_reconciliation_report_cmr_limit = 10
  create_reconciliation_report_cmr_page_size = 5
}
* Check environment variable for prefix-CreateReconciliationReport lambda, it should have CMR_LIMIT = 10 and CMR_PAGE_SIZE = 5
* Create an inventory report, and verify that collections only in CMR are listed and > 10.

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: example/spec/parallel/distributionAPI/distributionTEASpec.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ const s3Data = [
   '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf.met',
 ];
 
-xdescribe('TEA Distribution API', () => {
+describe('TEA Distribution API', () => {
```
Comment (jennyhliu) [inline example/spec/parallel/distributionAPI/distributionTEASpec.js:27]: not related to this ticket.  But TEA is still deployed and support, and we want to enable the test

File: docs/configuration/task-configuration.md [hunk 2]
```diff
@@ -6,6 +6,30 @@ hide_title: false
...
+- create_reconciliation_report_cmr_limit
+- create_reconciliation_report_cmr_page_size
+
+### Example
+
+```tf
+cmr_search_client_config = {
+  create_reconciliation_report_cmr_limit = 2500
+  create_reconciliation_report_cmr_page_size = 250
+}
```
Comment (flamingbear) [inline docs/configuration/task-configuration.md:30]: I think we should have a note or a link to the relevant CMR documentation of these variables and include that limit must be larger than page_size?

**Review by flamingbear — CHANGES_REQUESTED** [review]
I have one change and a few questions, happy to go over them in person if you want.  Looks good though.

File: packages/api/lambdas/create-reconciliation-report.js [hunk 3]
```diff
@@ -287,7 +314,7 @@ async function reconciliationReportForCollections(recReportParams) {
     if (!oneWayReport) collectionsOnlyInCmr = collectionsOnlyInCmr.concat(cmrCollectionIds);
   } catch (error) {
     log.error(`Error caught in reconciliationReportForCollections. with params ${JSON.stringify(recReportParams)}`);
-    log.error(errorify(error));
+    log.error(JSON.stringify(error));
```
[the remark this answers is not in the document]
Comment (jennyhliu) [reply packages/api/lambdas/create-reconciliation-report.js:317]: With `errorify`, deep object properties which are not part of the getOwnPropertyNames(error) are ignored. Such as `meta` isn't printed out with `errorify`.

with stringify(error):
```
{
    "level": "error",
    "message": "{\"name\":\"ResponseError\",\"meta\":{\"body\":{\"error\":{\"root_cause\":[{\"type\":\"search_context_missing_exception\",\"reason\":\"No search context found for id [10256]\"},{\"type\":\"search_context_missing_exception\",\"reason\":\"No search context found for id [10257]\"}],\"type\":\"search_phase_execution_exception\",\"reason\":\"all shards failed\",\"phase\":\"query\",\"grouped\":true,\"failed_shards\":[{\"shard\":-1,\"index\":null,\"reason\":{\"type\":\"search_context_missing_exception\",\"reason\":\"No search context found for id [10256]\"}},{\"shard\":-1,\"index\":null,\"reason\":{\"type\":\"search_context_missing_exception\",\"reason\":\"No search context found for id [10257]\"}}],\"caused_by\":{\"type\":\"search_context_missing_exception\",\"reason\":\"No search context found for id [10257]\"}},\"status\":404},\"statusCode\":404,\"headers\":{\"date\":\"Fri, 05 Nov 2021 18:21:27 GMT\",\"content-type\":\"application/json; charset=UTF-8\",\"content-length\":\"711\",\"connection\":\"keep-alive\",\"access-control-allow-origin\":\"*\"},\"warnings\":null,\"meta\":{\"context\":null,\"request\":{\"params\":{\"method\":\"GET\",\"path\":\"/_search/scroll/DnF1ZXJ5VGhlbkZldGNoAgAAAAAAACgQFlNkdlJuVjM3UjdLLWhOZjd2TlhmbEEAAAAAAAAoERZTZHZSblYzN1I3Sy1oTmY3dk5YZmxB\",\"body\":\"\",\"querystring\":\"scroll=3m\",\"headers\":{\"User-Agent\":\"elasticsearch-js/5.6.22 (linux 4.14.248-189.473.amzn2.x86_64-x64; Node.js v12.22.7)\",\"Content-Length\":\"0\"},\"timeout\":50000},\"options\":{\"warnings\":null},\"id\":2},\"name\":\"elasticsearch-js\",\"connection\":{\"url\":\"https://vpc-jl-tf-es-vpc-2lkv2q63z7rln5syilzh4hvnjy.us-east-1.es.amazonaws.com/\",\"id\":\"https://vpc-jl-tf-es-vpc-2lkv2q63z7rln5syilzh4hvnjy.us-east-1.es.amazonaws.com/\",\"headers\":null,\"deadCount\":0,\"resurrectTimeout\":0,\"_openRequests\":0,\"status\":\"alive\",\"roles\":{\"master\":true,\"data\":true,\"ingest\":true,\"ml\":false}},\"attempts\":0,\"aborted\":false}}}",
    "sender": "@api/lambdas/create-reconciliation-report",
    "timestamp": "2021-11-05T18:21:27.034Z"
}
```
with errorify(error):
```
{
    "level": "error",
    "message": "{\"stack\":\"ResponseError: search_phase_execution_exception\\n    at IncomingMessage.<anonymous> (/home/task/lambda-function/es-client/node_modules/@elastic/elasticsearch/lib/Transport.js:289:1)\\n    at IncomingMessage.emit (events.js:326:22)\\n    at IncomingMessage.EventEmitter.emit (domain.js:483:12)\\n    at endReadableNT (_stream_readable.js:1241:12)\\n    at processTicksAndRejections (internal/process/task_queues.js:84:21)\",\"message\":\"search_phase_execution_exception\",\"name\":\"ResponseError\",\"meta\":{\"meta\":{\"name\":\"elasticsearch-js\"}}}",
    "sender": "@api/lambdas/create-reconciliation-report",
    "timestamp": "2021-11-05T18:21:28.239Z"
}
```

File: example/deployments/cumulus/sandbox.tfvars [hunk 4]
```diff
@@ -32,10 +32,20 @@ s3_replicator_config = {
...
+  create_reconciliation_report_cmr_limit = 1500
+  create_reconciliation_report_cmr_page_size = 250
+}
+
 csdap_client_id = "csdap client id "
 csdap_client_password = "csdap client password"
 csdap_host_url = "https://auth.csdap.uat.earthdatacloud.nasa.gov"
 
+elasticsearch_client_config = {
+  create_reconciliation_report_es_scroll_duration = "8m"
```
[the remark this answers is not in the document]
Comment (jennyhliu) [reply example/deployments/cumulus/sandbox.tfvars:45]: I just wanted to add a working example of the configuration which does not use the default values.

File: packages/api/lambdas/create-reconciliation-report.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (flamingbear) [reply packages/api/lambdas/create-reconciliation-report.js]: It seems totally different, but I verified it does the same thing.  Nice.

**Review by flamingbear — APPROVED** [review]

**Comment by flamingbear** [comment]
Nice. LGTM

**Pull request merged**